src/streamlink/plugins/bongacams.py

```diff
@@ -0,0 +1,145 @@
+"""Streamlink plugin for bongacams.com"""
+
+import json
+import re
+import requests
+import time
+
+from io import BytesIO
+from hashlib import md5
+from urllib.parse import urljoin, urlsplit
+
+from streamlink.exceptions import PluginError, NoStreamsError
+from streamlink.packages.flashmedia.types import AMF0Value
+from streamlink.packages.flashmedia import AMFPacket, AMFMessage
+from streamlink.plugin.api import validate
+from streamlink.plugin import Plugin
+from streamlink.stream import RTMPStream
+from streamlink.utils import parse_json
+
+
+CONST_FLASH_VER = "WIN 24,0,0,186"
+CONST_DEFAULT_SWF_LOCATION = '/swf/chat/BCamChat.swf?cache=20161226150'
+CONST_AMF_GATEWAY_LOCATION = '/tools/amf.php'
+CONST_AMF_GATEWAY_PARAM = 'x-country'
+
+CONST_HEADERS = {}
+CONST_HEADERS['User-Agent'] = 'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) '
+CONST_HEADERS['User-Agent'] += 'Chrome/54.0.2840.99 Safari/537.36 OPR/41.0.2353.69'
+
+url_re = re.compile("^(https?://\w{2}.bongacams.com)/([\w\d_-]+)$")
+swf_re = re.compile("/swf/\w+/\w+.swf\?cache=\d+")
+
+amf_msg_schema = validate.Schema({
+    "status" : "success",
+    "userData" : {
+        "username": validate.text
+    },
+    "localData" : {
+        "NC_ConnUrl" : validate.url(scheme="rtmp"),
+        "NC_AccessKey" : validate.length(32),
+        "dataKey" : validate.length(32),
+    },
+    "performerData" : {
+        "username" : validate.text,
+    }
+})
+
+
+class AMFMessage2(AMFMessage):
+    @property
+    def size(self):
+        size = AMF0Value.size(self.value)
+        return size
+
+
+class bongacams(Plugin):
+    @classmethod
+    def can_handle_url(self, url):
+        return url_re.match(url)
+
+    def _get_stream_uid(self, username):
+        m = md5(username.encode('utf-8') + str(time.time()).encode('utf-8'))
+        return m.hexdigest()
+
+    def _get_streams(self):
+        match = url_re.match(self.url)
+
+        baseurl = match.group(1)
+        chathost = match.group(2)
+        chathost_url = match.group()
+        amf_gateway_url = urljoin(baseurl, CONST_AMF_GATEWAY_LOCATION)
+        country = urlsplit(baseurl).netloc[:2].lower()
+
+        # create http session and set headers
+        http_session = requests.Session()
```

Comment:
You should use `from streamlink.plugin.api import http` instead of creating a Session as it uses global options.